### ISSUE TITLE
Have problem when npm run build-prod

### ISSUE THREAD

**Issue by egnashko** [atais/ng2-eonasdan-datetimepicker#18]
Hi, it's okay with it in development mode but have ERROR in prod mode:
```
ERROR in Unexpected value 'DateTimePickerDirective in .../node_modules/ng2-eonasdan-datetimepicker/dist/datetimepicker.directive.d.ts' declared by the module 'AppModule in .../src/app/app.module.ts'

```
can you help me with it?

**Comment by iturn** [comment]
Maybe related to AOT compiling and loading, what is the result if you run the ng build without aot?
ng build --prod --aot=false

**Comment by egnashko** [comment]
With such command its okay, it works and build correctly, but i have command "npm run build-prod" in deployment, how can i use this with deploying? 
Also errors still exist wit "ng --prod" command.

**Comment by iturn** [comment]
this means, that the module is not AOT compatible. This is something the module owner could fix. 

You can do a ng build --prod  --aot=false and not have AOT in your production build. The normal ng build --prod command has default AOT set to true, that is why you get the errors. 

If you want AOT you could remove this module from your application and consider http://valor-software.com/ngx-bootstrap/#/datepicker

**Comment by atais** [comment]
What this module lacks for AOT support?

**Comment by iturn** [comment]
this could help https://medium.com/@isaacplmann/getting-your-angular-2-library-ready-for-aot-90d1347bcad

**Comment by egnashko** [comment]
I can't use the ngx-bootstrap datepicker because i need the date and time picker in one window, so thats why i decide to use this picker.

**Comment by iturn** [comment]
Ok then maybe make the changes in de module and create a PR for the owner

**Comment by atais** [comment]
@egnashko If you need this AOT change fast I would be really glad to accept the PR :+1:

**Comment by atais** [comment]
@iturn do you have any experience if the changes you mentioned (to adapt to AOT) may break the build with JIT?

**Comment by iturn** [comment]
Sorry, i dont know about that. But from what i read it should not. 
Also i read the first answer here, it might be very easy to provide AOT support.
https://stackoverflow.com/questions/38932193/angular-2-ahead-of-time-compilation-how-to

**Comment by egnashko** [comment]
@iturn so as i understand if i want to built my project with AOT it's better to use other date-time-picker?

**Comment by egnashko** [comment]
@atais i am sorry, i am new in code =) and don't know what "PR" means. What it means?

**Comment by iturn** [comment]
This module can have AOT fixed. This might be done by you or the owner of
the module.
A PR is a pull request, this means you contribute to the packackge, but
since you’re new in coding this seems unlikely.


Kind Regards,
Met vriendelijke groeten

Stijn Post

+31 6 38 75 46 44
www.iturn.it
stijn@iturn.it


On 27 July 2017 at 11:18:51, Yegor (notifications@github.com) wrote:

@iturn <https://github.com/iturn> so as i understand if i want to built my
project with AOT it's better to use other date-time-picker?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/atais/ng2-eonasdan-datetimepicker/issues/18#issuecomment-318307347>,
or mute the thread

**Comment by egnashko** [comment]
The code i using:
app.module.ts (uses DateTimePickerDirective cose without it have error that "A2Edatetimepicker is not ngModule")
```
...
import { DateTimePickerDirective } from 'ng2-eonasdan-datetimepicker/dist/datetimepicker.directive';
...
DateTimePickerDirective
```
main.ts
```
import 'eonasdan-bootstrap-datetimepicker';
```
angular.cli
```
styles: [
"../node_modules/eonasdan-bootstrap-datetimepicker/build/css/bootstrap-datetimepicker.min.css"
]
```
component html
```
<div class="input-group">
        <input class="form-control"
               type="text"
               placeholder="From"
               [(ngModel)]="filter.from"
               a2e-datetimepicker
               [date]="filter.from"
               (onChange)="dateChangeFrom($event)"
        />
        <span class="input-group-addon"><span class="glyphicon glyphicon-calendar"></span></span>
    </div>
```
component.ts
```
private a2eOptions = {format: 'MM/DD/YYYY hh:mm A'};

    dateChangeFrom(date) {
        this.filter.from = date.format(this.a2eOptions.format);
    }
    dateChangeTo(date) {
        this.filter.to = date.format(this.a2eOptions.format);
    }
```

**Comment by atais** [comment]
@egnashko If you are not able to fix this library you may either use other library or turn of AOT - until I do it or somebody else does.
Currently, I am busy with other projects so I can't promise this fix in the nearest future.

**Comment by egnashko** [comment]
@atais okay, thank you, i'll try to fix it and if it'll be success i'll write here. Thank you. Also if i couldn't do it i will wait you fixes in anytime.

**Comment by atais** [comment]
it should be fixed with #31 
right @fetrarij ?

**Comment by fetrarij** [comment]
@atais that's right, with  #31
